### PR TITLE
added alert aka toast component

### PR DESCRIPTION
**Pull request by ashishnnnnn**
-> Made A Toast Context Using which we can add toast(alert) in any part of page
-> Also added Toast Component.

### CODE REVIEW

File: src/Components/Product_card/Product_card.jsx [hunk 1]
```diff
@@ -16,6 +18,10 @@ export const Product_card = ({ item }) => {
         {is_item_in_wishlist(user_data.wishlist, item) ? (
           <a
```
Comment (lannisterrr) [inline src/Components/Product_card/Product_card.jsx]: what is the point of using "a" tag here? you can also use  a simple div.

File: src/Components/Product_card/Product_card.jsx [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import "./Product_card.css";
 import { useUserData } from "../../context/UserDataContext";
```
Comment (lannisterrr) [inline src/Components/Product_card/Product_card.jsx:2]: use **Barell imports** from one file.

**Review by lannisterrr — COMMENTED** [review]
Good work of context . Some suggesstions made.

File: src/Components/Toast/Toast.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      handleclosetoast();
+    }, 1500);
+    return () => {
+      clearTimeout(time_to_show_toast);
+    };
+  });
+  return (
+    <div className={`alert ${type}`}>
+      <p className="alert-text">{message}</p>
+      <a onClick={handleclosetoast} className="alert-cross">
```
Comment (ArjunGTX) [inline src/Components/Toast/Toast.jsx]: As suggested above `a` tag is not the best option here. It causes a page refresh unnecessarily

File: src/context/ToastContext.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  return (
+    <ToastContext.Provider value={{ toasts, handleaddtoast, handleclosetoast }}>
+      {children}
+
+      <ul
+        className={`alert-container ${toasts.length > 0 ? "" : "display-none"}`}
+      >
+        {toasts.map(({ id, message, type }) => (
+          <li key={id} className="mt-1">
+            <Toast
```
Comment (ArjunGTX) [inline src/context/ToastContext.js:36]: Liked the way you used `Toast` component

**Review by ArjunGTX — APPROVED** [review]

File: src/Components/Toast/Toast.jsx (hunk 3 above)
Comment (ashishnnnnn) [reply src/Components/Toast/Toast.jsx]: Done

File: src/Components/Product_card/Product_card.jsx (hunk 1 above)
Comment (ashishnnnnn) [reply src/Components/Product_card/Product_card.jsx]: Done

**Pull request merged**